Добавлен MailKit

BannikovRuslan opened the pull request:
Исправлено с учётом замечаний

BannikovRuslan committed:
Исправлено. Реализован интерфейс IEmailSender - описывающий метод отправки сообщений, класс MailKitEmailSender (IEmailSender), внутри которого реализуется логика отправки сообщений через MailKit. Также добавлена возможность прикрепления файла.

AntonIV8 commented on src/AsyncOpenXmlReportsSample/ODataBackend/Startup.cs:
```diff
@@ -82,6 +82,8 @@ public void ConfigureServices(IServiceCollection services)
                 .AddHealthChecks()
                 .AddNpgSql(connStr);
+
+            services.Configure<MailConfigurations.EmailOptions>(this.Configuration.GetSection("Email"));
```
А зачем регистрация конфигурации в стандартном контейнере ?
Вроде можно в методе ConfigureContainer все разрулить (без .Bind(emailOptions);)

AntonIV8 commented on src/AsyncOpenXmlReportsSample/ODataBackend/Templates/MailTemplates/T4MailTemplate.tt:
```diff
@@ -0,0 +1,64 @@
+﻿<#@ template language="C#" #>
```
А как используется этот шаблон ? Где указывается, что его применяем ?

AntonIV8 commented on src/AsyncOpenXmlReportsSample/ODataBackend/Templates/MailTemplates/T4MailTemplateAttachments.cs:
```diff
@@ -0,0 +1,21 @@
+﻿namespace IIS.AsyncOpenXmlReportsSample.Templates.MailTemplates
```
Этот класс не нужно. Про присоединение картинок написать в доке

AntonIV8 commented on src/AsyncOpenXmlReportsSample/ODataBackend/Templates/MailTemplates/T4MailTemplateData.cs:
```diff
@@ -0,0 +1,65 @@
+﻿
+namespace IIS.AsyncOpenXmlReportsSample.Templates.MailTemplates
```
T4MailTemplate.cs это генерируемый по tt класс. К таким классам, насколько мне кажется, нельзя дописывать что-то свое.
Если мы хотим поменять приветствие или подпись, то мы меняем это в tt.

BannikovRuslan committed:
Правки по замечаниям:
1. Убрана регистрация в стандартном контейнере
2. Добавлен отдельно метод отправки письма по шаблону Т4.
3. Удален класс с изображениями, используемых по умолчанию для шаблона Т4.
4. Переименованы классы, связанные с шаблоном Т4.

AntonIV8 commented on src/AsyncOpenXmlReportsSample/ODataBackend/AsyncOpenXmlReportsSample.ODataBackend.csproj:
```diff
@@ -8,10 +8,20 @@
     <XunitStartupFullName>IIS.AsyncOpenXmlReportsSample.Startup</XunitStartupFullName>
   </PropertyGroup>
+  <ItemGroup>
+    <None Remove="Templates\MailTemplates\T4MailTemplate.tt" />
```
а это что за Remove. Это он такую строчку после чего добавил ?

AntonIV8 commented on src/AsyncOpenXmlReportsSample/ODataBackend/Services/MailKitEmailService.cs:
```diff
@@ -0,0 +1,127 @@
+            {
+                Subject = subject,
+                HtmlMessage = body,
+                Greetings = $"Доброго времени суток, мой юный друг!",
```
Это нужно вынести в параметр метода, а то выходит сервис всегда будет отправлять сообщение с таким приветствием

AntonIV8 commented on src/AsyncOpenXmlReportsSample/ODataBackend/Services/MailKitEmailService.cs:
```diff
@@ -0,0 +1,127 @@
+            _emailOptions = emailOptions;
+        }
+
+        public void SendT4Email(string from, string to, string copyTo, string subject, string body, Dictionary<string, string> bodyAttachments, string fileName, byte[] fileBody)
```
Комментарии к методам, можно из интерфейса скопировать

BannikovRuslan committed:
Merge remote-tracking branch 'origin/main' into feature-231893-MailKit-Service

BannikovRuslan committed:
fix: внесены исправления по замечаниям

Удалены лишние связанные параметры Т4 шаблона  письма.

Добавлены комментарии к методам.

BannikovRuslan committed:
fix: переименован файл
